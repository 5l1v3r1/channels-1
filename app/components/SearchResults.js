// @flow
import React from 'react';
import Result from './Result';
import Loader from './Loader';
import BlankState from './BlankState';
import ErrorState from './ErrorState';
import InfiniteScroll from './InfiniteScroll';
import VideoType from '../dataTypes/videoType';
import styles from './SearchResults.scss';

const SearchResults = ({
  results,
  isFetching,
  error,
  pageToken,
  fetchSearch,
  searchResultsIsOpen
}: {
  results: VideoType[],
  isFetching: boolean,
  error: boolean,
  pageToken: string,
  fetchSearch: () => Promise<any>,
  searchResultsIsOpen: boolean
}) => {
  if ((isFetching && !pageToken) || error) {
    return (
      <div className={searchResultsIsOpen ? `${styles.searchResults} ${styles['searchResults--isOpen']}` : styles.searchResults}>
        {isFetching ? (
          <div className={styles.loader}>
            <Loader />
          </div>
        ) : (
          <div className={styles.errorState}>
            <ErrorState
              message={'Error requesting videos.'}
            />
          </div>
        )}
      </div>
    );
  }

  return (
    <div className={searchResultsIsOpen ? `${styles.searchResults} ${styles['searchResults--isOpen']}` : styles.searchResults}>
      {results.length ? (
        <InfiniteScroll isFetching={isFetching} pageToken={pageToken} loadMore={fetchSearch}>
          <ul className={styles.results}>
            {results.map(result => (
              <Result
                key={result.id}
                title={result.title}
                thumbnail={result.thumbnail}
                publishedAt={result.publishedAt}
                viewCount={result.viewCount}
              />
            ))}
          </ul>
        </InfiniteScroll>
      ) : (
        <div className={styles.blankState}>
          <BlankState
            message={'No videos found.'}
          />
        </div>
      )}
    </div>
  );
};

export default SearchResults;
