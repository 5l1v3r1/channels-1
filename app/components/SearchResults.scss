.searchResults {
  margin: 0;
  padding: 0;
  list-style: none;
}

.window {
  min-height: 98px;
  border-radius: 0 0 3px 3px;
}

.closeTarget {
  z-index: -1;
}
