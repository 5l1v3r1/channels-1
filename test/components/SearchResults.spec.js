import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchResults from '../../app/components/SearchResults';

function setup() {
  const props = {
    results: [],
    isFetching: false,
    error: false,
    pageToken: '',
    fetchSearch: () => jest.fn(),
    searchResultsIsOpen: false,
    handleSelect: () => jest.fn()
  };

  const enzymeWrapper = shallow(<SearchResults {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe('SearchResults', () => {
  describe('when closed', () => {
    it('should render self', () => {
      const { enzymeWrapper } = setup();
      const tree = toJson(enzymeWrapper);

      expect(tree).toMatchSnapshot();
    });
  });

  describe('when opened', () => {
    const { props, enzymeWrapper } = setup();
    beforeEach(() => {
      enzymeWrapper.setProps({
        searchResultsIsOpen: true
      });
    });

    afterEach(() => {
      enzymeWrapper.setProps({
        ...props
      });
    });

    it('should render self with loader', () => {
      enzymeWrapper.setProps({
        isFetching: true
      });
      const tree = toJson(enzymeWrapper);

      expect(tree).toMatchSnapshot();
    });

    it('should render self with error message', () => {
      enzymeWrapper.setProps({
        error: true
      });
      const tree = toJson(enzymeWrapper);

      expect(tree).toMatchSnapshot();
    });

    it('should render self with results', () => {
      enzymeWrapper.setProps({
        results: [
          {
            id: '4y_G6BayFz0',
            publishedAt: 'a year ago',
            thumbnail: 'https://i.ytimg.com/hqdefault.jpg',
            title: 'Professional Advice from Adobe MAX Conference',
            viewCount: '5,463'
          },
          {
            id: 'Flze-rwT7lM',
            publishedAt: '7 months ago',
            thumbnail: 'https://i.ytimg.com/vi/Flze-rwT7lM/hqdefault.jpg',
            title: 'If You Want to Survive in Design, You Better Learn to Code',
            viewCount: '39,430'
          }
        ]
      });
      const tree = toJson(enzymeWrapper);

      expect(tree).toMatchSnapshot();
    });

    it('should render self with no results', () => {
      const tree = toJson(enzymeWrapper);

      expect(tree).toMatchSnapshot();
    });
  });
});
