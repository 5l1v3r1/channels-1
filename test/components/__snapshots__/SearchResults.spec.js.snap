// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults should render self with error 1`] = `
<div
  className="searchResults isOpen"
>
  <ErrorState
    color="black"
    message="Error requesting videos."
  />
</div>
`;

exports[`SearchResults should render self with loader 1`] = `
<div
  className="searchResults isOpen"
>
  <Loader />
</div>
`;

exports[`SearchResults should render self with no results 1`] = `
<div
  className="searchResults isOpen"
>
  <BlankState
    color="black"
    message="No videos found."
  />
</div>
`;

exports[`SearchResults should render self with results 1`] = `
<div
  className="searchResults isOpen"
>
  <InfiniteScroll
    isFetching={false}
    loadMore={[MockFunction]}
    maxHeight="350px"
    pageToken="token"
  >
    <ul
      className="results"
    >
      <Video
        description=""
        handleClick={[Function]}
        isSearchResult={true}
        key="1"
        publishedAt="publishedAt"
        thumbnail="thumbnail.jpg"
        title="title"
        viewCount="1"
      />
      <Video
        description=""
        handleClick={[Function]}
        isSearchResult={true}
        key="2"
        publishedAt="publishedAt"
        thumbnail="thumbnail.jpg"
        title="title"
        viewCount="1"
      />
    </ul>
  </InfiniteScroll>
</div>
`;

exports[`SearchResults should render self with results 2`] = `
<div
  className="searchResults isOpen"
>
  <InfiniteScroll
    isFetching={true}
    loadMore={[MockFunction]}
    maxHeight="350px"
    pageToken="token"
  >
    <ul
      className="results"
    >
      <Video
        description=""
        handleClick={[Function]}
        isSearchResult={true}
        key="1"
        publishedAt="publishedAt"
        thumbnail="thumbnail.jpg"
        title="title"
        viewCount="1"
      />
      <Video
        description=""
        handleClick={[Function]}
        isSearchResult={true}
        key="2"
        publishedAt="publishedAt"
        thumbnail="thumbnail.jpg"
        title="title"
        viewCount="1"
      />
    </ul>
  </InfiniteScroll>
</div>
`;
